protocols/upnp/src/behaviour.rs

```diff
@@ -185,8 +199,32 @@ impl MappingList {
                             "could not request port mapping for multiaddress on the gateway: {}",
                             err
                         );
+                    } else {
+                        *state = MappingState::Pending;
+                    }
+                }
+                MappingState::Failed {
+                    retry_count,
+                    next_retry,
+                } => {
+                    if let Some(delay) = next_retry {
+                        if Pin::new(delay).poll(cx).is_ready() {
+                            let duration = MAPPING_DURATION;
+                            if let Err(err) = gateway.sender.try_send(GatewayRequest::AddMapping {
+                                mapping: mapping.clone(),
+                                duration,
+                            }) {
+                                tracing::debug!(
+                                    multiaddress=%mapping.multiaddr,
+                                    retry_count=%retry_count,
+                                    "could not retry port mapping for multiaddress on the gateway: {}",
+                                    err
+                                );
+                            } else {
+                                *state = MappingState::Pending;
```

Comment:
@RolandSherwin `state` is updated from `Failed` to `Pending` here. This could cause an infinite retry loop because `retry_count` gets lost?

Reply:
Thanks for pointing this out, fixed it now!